pkg/polaris/aws/aws.go

```diff
@@ -332,9 +334,51 @@ func (a API) AddAccount(ctx context.Context, account AccountFunc, features []cor
 		return uuid.Nil, fmt.Errorf("failed to get account: %s", err)
 	}
 
+	hasOutpostFeature := slices.ContainsFunc(
+		features, func(f core.Feature) bool {
+			return f.Equal(core.FeatureOutpost)
+		},
+	)
+
 	if config.config != nil {
+
+		hasOtherFeatures := slices.ContainsFunc(
+			features, func(f core.Feature) bool {
+				return !f.Equal(core.FeatureOutpost)
+			},
+		)
+
+		if hasOutpostFeature {
+			if options.outpostAccountID == "" {
+				return uuid.Nil, errors.New("outpost account id is not allowed to be empty")
+			}
+
+			// Default to using the config account
+			outpostConfig := config
+
+			if options.outpostAccountProfile != nil {
+				var err error
+				outpostConfig, err = options.outpostAccountProfile(ctx)
+				if err != nil {
+					return uuid.Nil, fmt.Errorf("failed to get outpost account: %s", err)
+				}
+			}
+			outpostConfig.id = options.outpostAccountID
+			outpostConfig.name = options.name
```

Comment:
Looking at the function code above, if we change this to:
```
outpostConfig.name = config.name
```
We should get the correct name, even if the user passes in something that doesn't match the original name used during the initial onboarding.